src/lib/utils/analytics.ts

```diff
@@ -65,15 +65,18 @@ export const formatSwapSignedAnalyticsEventProperties = ({
   allowedSlippage,
   fiatValues,
   txHash,
+  timeToSignSinceRequestMs,
 }: {
   trade: InterfaceTrade
   allowedSlippage: Percent
   fiatValues: { amountIn?: number; amountOut?: number }
   txHash?: string
+  timeToSignSinceRequestMs?: number
 }) => ({
   transaction_hash: txHash,
   token_in_amount_usd: fiatValues.amountIn,
   token_out_amount_usd: fiatValues.amountOut,
+  time_to_sign_since_request_ms: timeToSignSinceRequestMs,
```

Comment:
nit: you could pass in `beforeSign` and centralize the `Date.now() - beforeSign` here